### ISSUE TITLE
local.ERROR: Cannot redirect to an empty URL. 

### ISSUE THREAD

**Issue by flyingferret** [herpaderpaldent/seat-discourse#33]
Been using the plugin for > year.   Today got reported that the forums were down. Looking at the logs we are getting 

`local.ERROR: Cannot redirect to an empty URL. `

Checked my .env varibles and they all look good (unchange)

Please help!

Stacktrace https://pastebin.com/LDXwJ2kg

```
| Vendor          | Package Name           | Installed Version |
| --------------- | ---------------------- | ----------------- |
| eveseat         | api                    | 3.0.8             |
| eveseat         | console                | 3.0.6             |
| eveseat         | eveapi                 | 3.0.22            |
| eveseat         | notifications          | 3.0.6             |
| eveseat         | services               | 3.0.11            |
| eveseat         | web                    | 3.0.28            |
| flyingferret    | seat-whtools           | 0.2.0             |
| herpaderpaldent | seat-groups            | 1.7.2             |
| kassie          | calendar               | 3.0.16            |
| warlof          | seat-connector         | 1.4.5             |
| warlof          | seat-discord-connector | 4.0.6             |
| warlof          | seat-teamspeak         | 4.1.2             |
```
SeAT Discourse version 11.0

**Comment by herpaderpaldent** [comment]
On my mobile but according to the errorstack its because of this line https://github.com/herpaderpaldent/seat-discourse/blob/0d41ae16ecb63205101bcd5f3320c051d4534eca/src/Http/Controllers/SsoController.php#L69

So the env variable seems to be empty ... are you using docker?

**Comment by flyingferret** [comment]
No we're not using a docker install,  think this was caused when fixing issues with another plugin the command `php artisan config:cache` was run.  But not sure on the repair, the .env file was untouched

**Comment by flyingferret** [comment]
Talking with other Devs have had the following conversation.


kalaik utama  12:14 PM
@Warlof Tutsimo folllowing running the command php artisan config:cache  it seems some of my env varibles have been cleared :S  and broken my discourse plugin.   Although the .env are untouched the line
return redirect()->away(env('DISCOURSE_URL')); 
is returning null....eeep

Warlof Tutsimo  12:15 PM
env shouldn't be call outside config files
12:15
especially since once config is cached, env is non longer used :stuck_out_tongue:

kalaik utama  12:15 PM
rodger

Warlof Tutsimo  12:15 PM
php artisan config:clear will drop existing cached config
12:16
yeah
12:16
it's reason why it has been changed long time ago in SeAT 3

kalaik utama  12:16 PM
sorry to be a pain , just trying to get it back working for my alliance and pulling my hair out
12:17
so if i run config:clear we should be back to square one ?

Warlof Tutsimo  12:17 PM
https://laravel.com/docs/8.x/configuration#configuration-caching
12:18
yeah, since there we no more cached configuration, dotenv will always reload data from $_ENV and .env file (edited) 

kalaik utama  12:18 PM
thank you will try that again now, and will put this note on the issue i have raised with that plugin

Warlof Tutsimo  12:19 PM
and for timeline, here is initial fix on web repo https://github.com/eveseat/web/commit/e21361ac7639dc03410dcb57f5442778246afd82
12:19
should also have a few in eveapi package and others

kalaik utama  12:19 PM
thanks pal

**Comment by herpaderpaldent** [comment]
Yeah fair, when i created the package 2 years ago i was fairly inexperienced so this one of these bugs. 

Using the resource warlof gave you, you could write a PR which would fasten the development significantly.

**Comment by herpaderpaldent** [comment]
allright i got a fix: are you able to change your composer.json to require it?

```
herpaderpaldent/seat-discourse: "dev-fix/evn"
``` 

should do it. I have no application running to test it myself therefore before merging and tagging the fix i need you to confirm its working.
